doc/Architecture.md

```diff
@@ -246,8 +246,8 @@ The records validator serves as an access control list for the virtual DeNS data
 The Records Validator datum functions as the key in our virtual database. To update or create a record, a DeNS user simply adjusts the reference for the class and name that they own. To delete a record, the user submits a transaction with an output datum where the `reference` field is `Nothing`.
 
 ```haskell
-data DeNSKey = DeNSKey 
-    { class :: Word32
+data RecordDatum = RecordDatum 
```

Comment:
Above, we already used `DeNSKey` to mean something else, so this proposes to change the name to `RecordDatum` so we don't shadow it.

Reply:
Also, pretty sure that `class` is 16 bits since the [RFC](https://www.ietf.org/rfc/rfc1035.txt) says:
![image](https://github.com/mlabs-haskell/DeNS/assets/27853460/9a116a5a-cbe3-48eb-84c9-9debc86ff388)

Reply:
I think this is to be determined, we already state in the document that we will think of the class as 1 for all the current records, and use the class field at will.

Reply:
Er the class field should be 16 bits, but it's not that simple. I can't remember _where_ I read this, but somewhere in one of the RFCs they decided to reserve half of the classes for something. I'll try to look into it, though I don't think it matters.